src/main/java/com/cloudbees/jenkins/plugins/bitbucket/hooks/PullRequestHookProcessor.java

```diff
@@ -23,33 +23,174 @@
             BitbucketPullRequestEvent pull;
             if (instanceType == BitbucketType.SERVER) {
                 pull = BitbucketServerWebhookPayload.pullRequestEventFromPayload(payload);
             } else {
                 pull = BitbucketCloudWebhookPayload.pullRequestEventFromPayload(payload);
             }
-
             if (pull != null) {
-                String owner = pull.getRepository().getOwnerName();
-                String repository = pull.getRepository().getRepositoryName();
+                SCMEvent.Type eventType;
+                switch (hookEvent) {
+                    case PULL_REQUEST_CREATED:
+                        eventType = SCMEvent.Type.CREATED;
+                        break;
+                    case PULL_REQUEST_DECLINED:
+                    case PULL_REQUEST_MERGED:
+                        eventType = SCMEvent.Type.REMOVED;
+                        break;
+                    default:
+                        eventType = SCMEvent.Type.UPDATED;
+                        break;
+                }
+                // assume updated as a catch-all type
+                SCMHeadEvent.fireNow(new SCMHeadEvent<BitbucketPullRequestEvent>(eventType, pull, origin) {
+                    @Override
+                    public boolean isMatch(@NonNull SCMNavigator navigator) {
+                        if (!(navigator instanceof BitbucketSCMNavigator)) {
+                            return false;
+                        }
+                        BitbucketSCMNavigator bbNav = (BitbucketSCMNavigator) navigator;
+                        if (!isBitbucketServerUrlMatch(bbNav.getBitbucketServerUrl())) {
+                            return false;
+                        }
+                        if (!bbNav.getRepoOwner().equalsIgnoreCase(getPayload().getRepository().getOwnerName())) {
+                            return false;
+                        }
+                        return true;
+                    }
+
+                    private boolean isBitbucketServerUrlMatch(String serverUrl) {
+                        if (serverUrl == null) {
+                            // this is a Bitbucket cloud navigator
+                            if (getPayload() instanceof BitbucketServerPullRequestEvent) {
+                                return false;
+                            }
+                        } else {
+                            // this is a Bitbucket server navigator
+                            if (getPayload() instanceof BitbucketCloudPullRequestEvent) {
+                                return false;
+                            }
+                            Map<String, BitbucketHref> links = getPayload().getRepository().getLinks();
+                            if (links != null && links.containsKey("self")) {
+                                try {
+                                    URI navUri = new URI(serverUrl);
+                                    URI evtUri = new URI(links.get("self").getHref());
```

Comment:
there is always "self" and if there isn't we just assume they are the same URL and let the validation prove us right or wrong... but if we have an URL and it is for a different hostname, well we know that is not a match so we can save the event handler the effort of verification